### DISCUSSION TITLE
Nightscout on Google Cloud suddenly stopped to work

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#2551] [Q&A]
I set up Nightscout on Google Cloud following instructions at https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud with FreeDNS before a few weeks. Everything was working very good. That was my second setup, and I did the first a few months ago, but with noip. Yesterday I updated the first setup, created second FreeDNS subdomain in the same account and connected to it. The first setup works, but after a few hours after finishing the first setup, with the second setup, the BG values are no longer sent to Nightscout. I rebooted the server in SSH. I also updated the script according to https://navid200.github.io/xDrip/docs/Nightscout/NS_SyncExecutables, but nothing changed. Then I rebooted the phone and AAPS started sending treatments to Nightscout, but not xDrip. I rebooted the server again in SSH and Nightscout is no longer available, but an empty page with this message: `502 Bad Gateway`. Previously, while Nightscout was visible but not getting BGs, in xDrip -> System Status this option was red: REST-API problem: `Unable to do REST API Upload: Unable to do REST API Upload to connect to... after 30000ms marking recod: failed`. Status in SSH shows everything to be ok. I am still not able to understand what it is about. Maybe the problem is that I created a second subdomain in FreeDNS, or something else... maybe one of you can help me?

**Comment by Navid200** [comment]
Please post a screenshot of the status page as it answers a lot of questions.

**Reply by dar-kod** [reply]
Here are screenshots of System Status -> Uploaders from xDrip and SSH Status.

![Screenshot_2022-12-12-23-54-47-829_com eveningoutpost dexdrip](https://user-images.githubusercontent.com/24446806/207173905-2d498ca8-69a8-400b-bc4d-3babdfae032c.jpg)
![image](https://user-images.githubusercontent.com/24446806/207174595-7e20666a-b075-4601-8251-61959f4dd03c.png)

**Comment by Navid200** [comment]
What do you see when you enter the hostname in a web browser?
Make sure you use the hostname you see when you select Hostname and password on the status page.

**Reply by dar-kod** [reply]
When I enter the hostname this is what I see:

![image](https://user-images.githubusercontent.com/24446806/207203552-d47f2a08-d8b9-4875-8909-04d0ed8274d4.png)

**Reply by Navid200** [reply]
Log into FreeDNS.
In the left pane, click on "Dynamic DNS".
At the bottom of the page, you should see your two hostnames each, in a row.  In each row, on the right, you can see the corresponding IP address.  Are the two IP addresses different or the same?

**Reply by Navid200** [reply]
By any chance, are your two subdomains identical except in capitalization?  
If yes, you shouldn't so that.

**Reply by dar-kod** [reply]
In "Dynamic DNS" two IP addresses are different.

Two subdomains are not identical, but they have identical domain.

**Comment by tzachi-dar** [comment]
@dar-kod I'll create a script that checks the status of nightscout and upload it here. This will give us more information about where the problem is.

**Reply by Navid200** [reply]
I don't  want his Nightscout not be operational for such a long time.
Can I ask him to run the FreeDNS utility again?

**Reply by dar-kod** [reply]
@Navid200 Before reading the comment of tzachi-dar I started the FreeDNS utility using first option 1, and since it did not lead to changes, then option 2. Nothing changed.

**Reply by Navid200** [reply]
What do you mean by starting the utility using first option 1?

You start the FreeDNS setup utility from the main menu using item number 10.

**Reply by dar-kod** [reply]
Yes, I started the FreeDNS setup utility again. 

I also created a third FreeDNS subdomain and ran the FreeDNS setup utility again, but nothing changed.

**Reply by Navid200** [reply]
This makes no sense.

You can backup your database:  https://navid200.github.io/xDrip/docs/Nightscout/DatabaseBackup.html
Ensure to download the backup onto your real (not virtual) computer.

Then, you can delete the virtual machine and create a new one going through the entire installation procedure:  https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html
This will completely avoid dealing with noip dot come.  

After everything is done and Nightscout working again, you can restore your database: https://navid200.github.io/xDrip/docs/Nightscout/DatabaseRestore.html

**Comment by tzachi-dar** [comment]
@dar-kod are you still getting the 502 error? I guess that there is a very good chance that Navid suggestion will solve the issue faster, but if that does not work for you I can write the script that I have promised.

**Reply by dar-kod** [reply]
@tzachi-dar I did what Navid suggested. Now, it is working. Thank you.

**Reply by Navid200** [reply]
Can you check the other Nightscout that you updated yesterday with a new FreeDNS hostname causing this problem?

Is the other Nightdscout working properly also?

**Reply by dar-kod** [reply]
@Navid200 The other Nightscout is working properly also.

**Comment by martingay1975** [comment]
Just to record I also got the same error, after seemingly no changes.  Same symptoms dar-kod. 

I fixed by going into Google Cloud -> Compute Engine -> clicking on the three dots next to my VM instance. I selected 'Stop'. Then once stopped (couple of minutes). I selected'Start/Resume'. It worked again. Switch off and on again basically. :)

**Reply by Nisarockstar** [reply]
I had the same, 502 Gateway error when trying to access Nightscout url, hadn't changed anything since initial build beginning of March.  Did the same as @martingay1975, Google Cloud -> Compute Engine -> three dots next to VM instance -> Stop -> wait a few minutes -> Start/Resume and the Nightscout url is working again and eventually uploaded the missing data from the last 3 days.

**Reply by Navid200** [reply]
The issue described in this discussion was resolved a long time ago.

If you do something that causes your ip address to change, you will need to restart the server, which will cause the ip to be mapped again to your hostname.

There are times you may see the 502 gateway disconnect issue during a platform update from a very old setup.  That is described with how to resolve here: https://navid200.github.io/xDrip/docs/Nightscout/502BadGateway.html
